diffpy/snmf/io.py

```diff
@@ -3,6 +3,9 @@
 from pathlib import Path
 from diffpy.utils.parsers.loaddata import loadData
 
+PDF_FILE_EXTENSIONS = {'.gr', '.cgr'}
+DIFFRACTION_FILE_EXTENSIONS = {'.chi', 'iq', 'xye', '.xrd', '.xy'}
```

Comment:
In a much earlier comment, you mentioned that finding the file type by the file extension may not be a good idea. Do you know of any libraries that can do this (such as in diffpy.utils).

Reply:
it looks to me as if you are trying to find the data_type from the extension.  Is it teh file-type you are after?  LoadData is pretty flexible, it should successfully read all the file extensions you put up there (I am not familiar with .xrd)

Reply:
in other words, I don;t think you need to do this.